civet.dev/cheatsheet.md

```diff
@@ -5,1638 +5,297 @@ aside: false
+    display: none
+  }
+</style>
+
+<div className="cheatsheet">
+
+```ts
+// Declarations
+const x = 1
+let y: number = 2
+var z: string = "3"
+x := 1 // const x = 1
+y .= 2 // let y = 2
+
+// Destructuring
+[ a, b ] := x
+[ ..., last ] := x
+[ first, ...rest ] := x
+[ first, ..., last ] := x
+[ first, ...middle, last] := x
+
+{ a, b } := c
 
-In most cases, the Civet code on
-<span class="wide">the left</span><span class="narrow">top</span>
-is optional shorthand.
-The TypeScript code on
-<span class="wide">the right</span><span class="narrow">bottom</span>
-(and most TypeScript code) is almost always also valid Civet input.
+```
 
-[[toc]]
+```ts
+// Function application
+f(x)
+f(a, g(x))
+f(...args, cb)
 
-## Variable Declaration
+// Implicit application
+f x // f(x)
+f a, b, c // f(a, b, c)
+f g x // f(g(x))
+f a, b c // f(a, b(c))
 
-By default, you are responsible for declaring your variables
-via `var`, `let`, `const`, or their shorthands:
+```
 
-<Playground>
-a := 10
-b .= 10
-c: number | string .= 0
-let d: boolean
-var v: any
-</Playground>
+```ts
+// Conditionals
+x && y
+x and y // x && y
+x || y
+x or y // x || y
+
+// relationals
+x === y
+x is y // x === y
+x < y
+x > y
+// Chained relationals
+x < y < z // x < y && y < z
+```
 
-Alternatively, you can use a `"civet"` directive at the beginning of your file
-to specify one of two automatic variable declaration modes:
+```ts
+// this
+this
+// shorthand
+@ // this
+@x // this.x
+
+// instance of and typeof shorthand
+x instanceof y
+x <? Y // x instanceof y
+typeof x === "string"
+x <? "string" // typeof x === "string"
+```
 
-### `autoVar`
+`````ts
+// strings
+"I'm a string"
+'I\'m also a string'
+`I'm a ${template} string`
 
-<Playground>
-"civet autoVar"
-sos = 0
-for item of iterable
-  square = item * item
-  sos += square
-</Playground>
+// Block Strings
+"""
+  Block strings
+  will dedent
+"""
 
-### `autoLet`
+'''
+  They work with all kinds of
+  strings
+'''
 
-<Playground>
-"civet autoLet"
-sos = 0
-for item of iterable
-  square = item * item
-  sos += square
-</Playground>
+```
+  I will dedent by removing
+  common indentation
+```
 
-### Declarations in Conditions and Loops
+`````
 
-<Playground>
-if match := regex.exec string
-  console.log match[1], match[2]
-</Playground>
+```ts
+// if conditions
+if x < 3
+  "it's small"
 
-<Playground>
-if [, dir, base] := /^(.*\/)?([^/]*)$/.exec file
-  console.log dir, base
-</Playground>
+if x > 11
+  "it's big"
 
-<Playground>
-if {x, y} := getLocation()
-  console.log `At ${x}, ${y}`
-else
-  console.log "Not anywhere"
-</Playground>
+unless paused
+  run()
 
-<Playground>
-node .= linkedList.head
-while {data, next} := node
-  console.log data
-  node = next
-</Playground>
+// loops
+while x < 10
+  f(x)
+  x++
 
-## Objects
+for item of items
+  update item
 
-### Unbraced Literals
+for key in object
+  log key
 
-When every property has a value, braces can be omitted.
+for own key in object
+  log `my ${key}`
+```
```

Comment:
I might add the two-binding forms of `for..of/in`. Not sure whether to modify the existing examples or add them separately.